portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,20 +27,23 @@
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
 
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int maxComments = Integer.parseInt(request.getParameter("max-comments"));
```

Comment:
It's also a good idea in general to set a maximum number of comments.

Right now I can crash your server by requesting more comments than your server can hold in memory (assuming there are actually that many in the database)